package.json

```diff
@@ -89,6 +89,7 @@
     "redux": "^3.5.2",
     "redux-thunk": "^2.1.0",
     "request": "^2.76.0",
+    "reselect": "^3.0.1",
```

Comment:
[reselect](https://github.com/reduxjs/reselect) is a small library for memoization. It is often paired with Redux to optimize selectors so that they only recompute when relevant parts of the application state have changed. Here it is used to optimize `store.allGroups()`.